### DISCUSSION TITLE
4:3 resolution options for webcam on Hero 10

### DISCUSSION THREAD

**Discussion by rosehendrix** [gopro/labs#1037] [Q&A]
Hi! It seems like only 16:9 outputs are supported for a streaming webcam. Is there a way to get 4:3 resolution options/is that supported/could it be supported? Eg 720x540 or 1440x1080. I am using the process as [here](https://github.com/jschmid1/gopro_as_webcam_on_linux) for accessing webcam functionality. Thanks for any assistance!

**Comment by dnewman-gpsw** [comment]
16:9 only to my knowledge.

**Comment by rosehendrix** [comment]
Awesome, thanks, that's what I thought. Is there a way to submit a feature request to support 4:3? It seems on the face of it like it shouldn't be a huge lift.

**Reply by dnewman-gpsw** [reply]
It is all incremental, one addition video mode sounds small, but the permutations grow, making QA more difficult. This is why from camera generation to generation, video modes are dropped, making way for new ones.

**Reply by rosehendrix** [reply]
Fair enough. Is there a released camera that does support 4:3 webcam resolution modes?

**Reply by dnewman-gpsw** [reply]
No, I don't think so.  My guess (I don't have any data) is that only a small percentage use the webcam mode, and that is what limits it to the one aspect ratio.

**Reply by rosehendrix** [reply]
Is there any possible path I could take (e.g. modifying the firmware myself, with the appropriate level of caveat emptor) in order to expose a 4:3 mode to webcam streaming? My goal is to get as much sensor information as is available, and my impression is that the 16:9 modes all crop the image.

**Reply by dnewman-gpsw** [reply]
If not possible to externally modified the firmware, all firmware is security signed. Even me, as a GoPro firmware developer can't make signed firmware on my local system.  That path is a non-starter.

However, webcam supports Superview which is 4:3 image presented within a 16:9 1080p image. So the full height of a 4:3 sensor is presented (optionally) to webcam.

**Reply by rosehendrix** [reply]
Okay, excellent. (Thank you for answering my questions!) So if I wanted to essentially reconstruct a 4:3 wide mode stream, would superview with a resize to unsquish it be equivalent? Or is there something more sophisticated going on that would be more difficult to unwind/replicate?

**Reply by dnewman-gpsw** [reply]
Superview is 4:3, the unsquish does involve math, but it is reversible. There are third party tool that undo it, some might be open source. Way, way easier than adding a new video mode.
